tools/labs/skels/kernel_api/6-list-sync/list-sync.c

```diff
@@ -60,19 +62,23 @@ static void task_info_add_to_list(int pid)
 	struct task_info *ti;
 
 	/* TODO 1: Protect list, is this read or write access? */
+	spin_lock(&lock);
 	ti = task_info_find_pid(pid);
 	if (ti != NULL) {
 		ti->timestamp = jiffies;
 		atomic_inc(&ti->count);
 		/* TODO: Guess why this comment was added  here */
-		return;
```

Comment:
You need to do spin_unlock here before exiting the function. Otherwise, you might get into a deadlock.